codersdk/workspaceagentconn.go

```diff
@@ -120,19 +126,64 @@ func init() {
 	}
 }
 
+// NewWorkspaceAgentConn creates a new WorkspaceAgentConn. `conn` may be unique
+// to the WorkspaceAgentConn, or it may be shared in the case of coderd. If the
+// conn is shared and closing it is undesirable, you may return ErrNoClose from
+// opts.CloseFunc. This will ensure the underlying conn is not closed.
+func NewWorkspaceAgentConn(conn *tailnet.Conn, opts WorkspaceAgentConnOptions) *WorkspaceAgentConn {
+	return &WorkspaceAgentConn{
+		Conn: conn,
+		opts: opts,
+	}
+}
```

Comment:
Should this all be in `agentsdk` instead? Seems like a better spot

Reply:
yeah i think that makes more sense

Reply:
This shouldn't go in agentsdk, agentsdk is for agent making requests to coderd. `WorkspaceAgentConn` is clients making connections to agents.